src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -157,51 +157,52 @@ protected void ProcessUrlAttribute(string attributeName, TagHelperOutput output)
                 throw new ArgumentNullException(nameof(output));
             }
 
-            IReadOnlyList<TagHelperAttribute> attributes;
-            if (output.Attributes.TryGetAttributes(attributeName, out attributes))
+            for (var i = 0; i < output.Attributes.Count; i++)
```

Comment:
Agree this largish-looking change was required. Might also improve performance because `TryGetAttributes()` allocates a `List<T>` in all non-empty cases.